bot/cogs/queue.py

```diff
@@ -178,78 +161,65 @@ async def join(self, ctx):
             title = f'Unable to add **{ctx.author.display_name}**: Their account is not linked'
         else:  # Message author is linked
             player = await self.bot.api_helper.get_player(ctx.author)
+            await self.bot.db_helper.insert_users(ctx.author)
+            queue_ids = await self.bot.db_helper.get_queued_users(ctx.guild)
+            queue = [self.bot.get_user(user_id) for user_id in queue_ids]
+            guild_data = await self.bot.db_helper.get_guild(ctx.guild)
```

Comment:
There are 4 awaited queries back to back here - you can probably run them in parallel. Or at least the ones that don't rely on each other

Reply:
I'll have to think more about how to do that. I'm planning on getting rid of the `get_user()` calls after this merges since it's not DB related. I timed it and it actually took 2 seconds to get 10 users from the Discord API.

Reply:
I had a quick google, could you use `asyncio.gather`? That might not be right I'm not sure :P

Reply:
`asyncio.gather()` seems like the right tool for the job. I will look into implementing that wherever we have independent helper calls.

Reply:
I implemented `asyncio.gather()` for the 4 queries/api calls and the time went from 33ms to 27ms. Not crazy but it's obviously better. This could have a much bigger impact with larger tables and multiple awaits that take a while.